### PR TITLE
WebAuthn and Simulator detection 

### PR DESCRIPTION
**Pull request by sparsh-otpless**

### CODE REVIEW

File: Example/OtplessSDK/LoggerVC.swift [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+//  Created by Sparsh on 26/07/24.
+//  Copyright © 2024 CocoaPods. All rights reserved.
+//
+
+import UIKit
+
+class LoggerVC: UIViewController {
+    var tableView: UITableView!
+    
+    var headerLabel: UILabel = {
```
Comment (digvijay-otpless) [inline Example/OtplessSDK/LoggerVC.swift:14]: initialize the headerLabel in `setupHeaderLabel` method same as it is getting done for `tableView`.

File: OtplessSDK/Classes/DeviceInfoUtils.swift [hunk 2]
```diff
@@ -154,6 +160,26 @@ class DeviceInfoUtils {
     func getTrackingSessionId() -> String? {
         return tsid
     }
+    
+    
+    /// Determines whether the device is simulator.
+    ///
+    /// - returns: Boolean indicating whether device is simulator or not. Returns true if the device is simulator, else false.
+    func isDeviceSimulator() -> Bool {
+        #if swift(>=4.1)
```
Comment (digvijay-otpless) [inline OtplessSDK/Classes/DeviceInfoUtils.swift:169]: shorten the code
 #if swift(>=4.1)
            return targetEnvironment(simulator)
        #else
           return (arch(i386) || arch(x86_64)) && os(iOS)
        #endif

File: OtplessSDK/Classes/OtplessView.swift [hunk 3]
```diff
@@ -221,10 +216,16 @@ class OtplessView: UIView {
...
+                if #available(iOS 16, *) {
+                    let queryWebAuthn = URLQueryItem(name: "isWebAuthnSupported", value: "true")
+                    urlComponents.queryItems?.append(queryWebAuthn)
+                }
+                
                 let updatedUrlComponents = addInitialParams(urlComponents: urlComponents)
 
                 if let updatedURL = updatedUrlComponents.url {
                     let request = URLRequest(url: updatedURL)
+                    OtplessLogger.log(string: request.url?.absoluteString ?? "Unable to get URL", type: "WebView URL")
```
Comment (dev-otpless-tech) [inline OtplessSDK/Classes/OtplessView.swift:228]: Remove this for release scheme

File: OtplessSDK/Classes/OtplessWebAuthnManagerImpl.swift [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+//
+//  Created by Sparsh on 27/06/24.
+//
+
+import Foundation
+import LocalAuthentication
+import AuthenticationServices
+
+@available(iOS 16.0, *)
+class OtplessWebAuthnManagerImpl: NSObject, OtplessWebAuthnManager, ASAuthorizationControllerDelegate, ASAuthorizationControllerPresentationContextProviding {
```
Comment (dev-otpless-tech) [inline OtplessSDK/Classes/OtplessWebAuthnManagerImpl.swift:13]: Can we move to extenstions of Otplesswebauthn

File: OtplessSDK/Classes/OtplessWebAuthnManager.swift [hunk 5]
```diff
@@ -0,0 +1,30 @@
+//
+//  OtplessWebAuthnManager.swift
```
Comment (dev-otpless-tech) [inline OtplessSDK/Classes/OtplessWebAuthnManager.swift:2]: not a manager make it Otplesswebauthn

File: OtplessSDK/Classes/OtplessWebAuthnManagerImpl.swift [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+    ///
+    /// - parameter request: The request dictionary containing registration parameters.
+    /// - parameter onResponse: The callback to handle the registration response.
+    func initiateRegistration(
+        withRequest request: [String : Any],
+        onResponse responseCallback: @escaping (WebAuthnResult) -> Void
+    ) {
+        self.responseCallback = responseCallback
+        
+        if !isWindowValid() {
```
Comment (dev-otpless-tech) [inline OtplessSDK/Classes/OtplessWebAuthnManagerImpl.swift:47]: do the consideration analysis

File: OtplessSDK/Classes/OtplessWebAuthnManagerImpl.swift [hunk 7]
```diff
@@ -0,0 +1,345 @@
...
+            responseCallback(.failure(errorJson))
+            return
+        }
+        
+        createRegistrationRequest(
+            from: request,
+            onErrorCallback: responseCallback,
+            onRegistrationRequestCreation: { platformKeyRequest in
+                let authController = ASAuthorizationController(authorizationRequests: [ platformKeyRequest ])
+                authController.delegate = self
```
Comment (dev-otpless-tech) [inline OtplessSDK/Classes/OtplessWebAuthnManagerImpl.swift:58]: explore the all function which can lead to success and failure and utilize them accordingly for feedback

File: OtplessSDK/Classes/OtplessWebAuthnManagerImpl.swift [hunk 8]
```diff
@@ -0,0 +1,345 @@
...
+            let errorJson = Utils.createErrorDictionary(error: "window_nil", errorDescription: "The view was not in the app's view hierarchy.")
+            responseCallback(.failure(errorJson))
+            return
+        }
+        
+        createSignInRequest(
+            from: request,
+            onErrorCallback: responseCallback,
+            onSignInRequestCreation: { platformKeyRequest in
+                let authController = ASAuthorizationController(authorizationRequests: [ platformKeyRequest ])
```
Comment (dev-otpless-tech) [inline OtplessSDK/Classes/OtplessWebAuthnManagerImpl.swift:86]: Common this piece of code

File: OtplessSDK/Classes/OtplessWebAuthnManagerImpl.swift [hunk 9]
```diff
@@ -0,0 +1,345 @@
...
+        } else {
+            // Some other authorization type was used like passwords.
+            let errorJson = Utils.createErrorDictionary(error: "unexpected_credential_type", errorDescription: "Unexpected credential type \(authorization.credential.description)")
+            responseCallback?(.failure(errorJson))
+        }
+    }
+}
+
+@available(iOS 16, *)
+extension OtplessWebAuthnManagerImpl {
```
Comment (dev-otpless-tech) [inline OtplessSDK/Classes/OtplessWebAuthnManagerImpl.swift:160]: change the name

File: OtplessSDK/Classes/OtplessWebListenerImpl.swift [hunk 10]
```diff
@@ -0,0 +1,321 @@
...
+//  OtplessWebManagerImpl.swift
+//  OtplessSDK
+//
+//  Created by Sparsh on 25/06/24.
+//
+
+import Foundation
+import WebKit
+
+class OtplessWebListenerImpl: OtplessWebListener {
```
Comment (dev-otpless-tech) [inline OtplessSDK/Classes/OtplessWebListenerImpl.swift:11]: do the changes as per discussion

**Review by dev-otpless-tech — CHANGES_REQUESTED** [review]

**Review by dev-otpless-tech — APPROVED** [review]

**Pull request merged**